src/requestSchemas.ts

```diff
@@ -0,0 +1,210 @@
+  'system-version': z.string()
+})
+  .partial()
+  .strict()
+  .superRefine(catchInvalidParams([catchMissingIdentifyingInfo], UNSUPPORTED_PACKAGE_ARGS));
+
+export const CommonDataRequirementsArgs = IdentifyingParameters.extend({
+  'check-system-version': z.string(),
+  'force-system-version': z.string(),
+  'include-components': stringToBool,
+  'include-dependencies': stringToBool,
+  manifest: z.string(),
+  parameters: z.string(),
+  periodEnd: checkDate('periodEnd'),
+  periodStart: checkDate('periodStart'),
+  'system-version': z.string()
+})
+  .partial()
+  .strict();
+
+export const MeasureDataRequirementsArgs = CommonDataRequirementsArgs.superRefine(
+  catchInvalidParams([catchMissingIdentifyingInfo], UNSUPPORTED_DATA_REQ_ARGS)
+);
+
+export const LibraryDataRequirementsArgs = CommonDataRequirementsArgs.extend({
+  expression: z.string()
+})
+  .partial()
+  .strict()
+  .superRefine(catchInvalidParams([catchMissingIdentifyingInfo], UNSUPPORTED_DATA_REQ_ARGS));
+
+export const CoreSearchArgs = z
+  .object({
+    'composed-of': z.string(),
+    context: z.string(),
+    'context-quantity': z.string(),
+    'context-type': z.string(),
+    'context-type-quantity': z.string(),
+    'context-type-value': z.string(),
+    date: checkDate('date'),
+    'depends-on': z.string(),
+    'derived-from': z.string(),
+    description: z.string(),
+    effective: z.string(),
+    identifier: z.string(),
+    jurisdiction: z.string(),
+    name: z.string(),
+    predecessor: z.string(),
+    status: z.string(),
+    successor: z.string(),
+    title: z.string(),
+    topic: z.string(),
+    url: z.string().url(),
+    version: z.string()
+  })
+  .partial()
+  .strict();
+
+export const LibrarySearchArgs = CoreSearchArgs.extend({
+  'content-type': z.string(),
+  type: z.string()
+})
+  .partial()
+  .strict()
+  .superRefine(catchInvalidParams([], UNSUPPORTED_LIBRARY_SEARCH_ARGS));
+
+export const MeasureSearchArgs = CoreSearchArgs.extend({
+  publisher: z.string()
+})
+  .partial()
+  .strict()
+  .superRefine(catchInvalidParams([], UNSUPPORTED_MEASURE_SEARCH_ARGS));
+
+/**
+ * wrapper around the Zod parse function that catches Zod errors, parses the messages and throws appropriate ServerErrors
+ */
+export function parseRequestSchema<T extends z.ZodTypeAny>(params: Record<string, any>, schema: T): z.infer<T> {
+  try {
+    return schema.parse(params);
+  } catch (e) {
+    if (e instanceof z.ZodError) {
+      if (e.issues[0].code === z.ZodIssueCode.custom) {
+        if (e.issues[0].params?.type === 'NotImplemented') {
+          throw new NotImplementedError(e.issues[0].message);
+        } else {
+          throw new BadRequestError(e.issues[0].message, e.issues[0].params?.serverErrorCode);
+        }
+      } else if (e.issues[0].code === z.ZodIssueCode.unrecognized_keys) {
+        throw new BadRequestError(e.issues[0].message, constants.ISSUE.CODE.VALUE);
+      }
+      throw new BadRequestError(e.issues[0].message);
```

Comment:
It looks like the default message for the incorrect type of a key is just "invalid input". Is there any way to make that better? I forget what they were like before. Maybe this is fine, but at the very least calling out the key that's in violation if that's available in the issue is a good call.

I sent `include-terminology=asdf` to violate the boolean schema for testing

Reply:
This might also be specific to the union case. For example when I pass in a bad url the output is "invalid url" which is correct info